src/components/expanded-state/unique-token/ZoomableWrapper.js

```diff
@@ -0,0 +1,623 @@
+import React from 'react';
+import { StatusBar, StyleSheet, View } from 'react-native';
+import {
+  PanGestureHandler,
+  PinchGestureHandler,
+  TapGestureHandler,
+} from 'react-native-gesture-handler';
+import Animated, {
+  Easing,
+  runOnJS,
+  useAnimatedGestureHandler,
+  useAnimatedStyle,
+  useSharedValue,
+  useWorkletCallback,
+  withSpring,
+  withTiming,
+} from 'react-native-reanimated';
+import styled from 'styled-components';
+import useReactiveSharedValue from '../../../react-native-animated-charts/src/helpers/useReactiveSharedValue';
+import { ButtonPressAnimation } from '../../animations';
+import { useDimensions } from '@rainbow-me/hooks';
+import { position } from '@rainbow-me/styles';
+
+const adjustConfig = {
+  duration: 300,
+  easing: Easing.bezier(0.4, 0, 0.22, 1),
+};
+const enterConfig = {
+  damping: 40,
+  mass: 1.5,
+  stiffness: 600,
+};
+const exitConfig = {
+  damping: 68,
+  mass: 2,
+  stiffness: 800,
+};
+const GestureBlocker = styled(View)`
+  height: ${({ height }) => height};
+  left: ${({ containerWidth, width }) => -(width - containerWidth) / 2};
+  position: absolute;
+  top: -85;
+  width: ${({ width }) => width};
+`;
+const Container = styled(Animated.View)`
+  align-self: center;
+  shadow-color: ${({ theme: { colors } }) => colors.shadowBlack};
+  shadow-offset: 0 20px;
+  shadow-opacity: 0.4;
+  shadow-radius: 30px;
+`;
+const ImageWrapper = styled(Animated.View)`
+  ${position.size('100%')};
+  flex-direction: row;
+  overflow: hidden;
+`;
+const ZoomContainer = styled(Animated.View)`
+  height: ${({ height }) => height};
+  width: ${({ width }) => width};
+`;
+
+const MAX_IMAGE_SCALE = 4;
+const MIN_IMAGE_SCALE = 1;
+const THRESHOLD = 250;
+
+export const ZoomableWrapper = ({
+  animationProgress: givenAnimationProgress,
+  children,
+  horizontalPadding,
+  aspectRatio,
+  isENS,
+  borderRadius,
+  disableAnimations,
+  yDisplacement: givenYDisplacement,
+}) => {
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const animationProgress = givenAnimationProgress || useSharedValue(0);
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const yDisplacement = givenYDisplacement || useSharedValue(0);
+
```

Comment:
I can see why are you doing this, I wonder if there's an alternative.

Reply:
any ideas? I have an opinion that this is actually good and the rules of hooks should be more lenient here.